README.md

```diff
@@ -52,4 +52,66 @@ cp configs/* /etc/redeem
 cp data/* /etc/redeem
 ```
 
-Other dependencies: [source install docs](https://intelligent-agent.github.io/redeem/replicape/redeem.html#from-source)
+### Source install with Python virtual environments
+
+Using Python `virtualenv` is the suggested installation method when running from source for anything more than casual development.  Virtual environments allow multiple, independent installs of Redeem that are isolated from each other and other applications on the system.
+
+The overview of the virtual structure includes directories in `/usr/local`, one for each working copy of Redeem that is desired.  The script for starting Redeem is modified to activate the selected virtual environment and start the service.
+
+Note that the `2.1.x` and later branches of redeem need `pip` version 10.  This setup is done as `root`.
+
+1. `cd /usr/local/lib/python2.7/dist-packages`
+
+2. Delete the installed Redeem python packages from this directory, but keep the OctoPrint plugin.  `rm Redeem* redeem*`
+
+3. `cd /usr/local`
+
+4. Create the virtual environment with `virtualenv 2.2`  This will create a directory named `2.2` that will be the root of that virtual environment.  The name given name, `2.2`, is for demonstration purposes, feel free to choose a different name.
+
+5. `cd 2.2`
+
+6. Allow the use of system libraries with `rm lib/python/2.7/no-global-site-packages.txt`
+
+7. Make the virtual environment active with `source bin/activate`.  This step isn't needed to edit files, but the virtual environment should be active when running `setup.py`.  In order to indicate that a virtual environment is active, the command prompt will have the environment name prefixed to the command prompt, like `(2.2) root@kamikaze:/usr/local/2.2`
+
+8. Create a directory for the git repository, and clone it.
+
+   ```
+   mkdir src
+   cd src
+   git clone https://github.com/intelligent-agent/redeem.git
+   cd redeem
+   git checkout origin/2.2.x
+   ```
+
+   
+
+9. Get things set up with `python setup.py develop`
+
+10. If needed, copy the configuration files `cp configs/* /etc/redeem` and/or `cp data/* /etc/redeem`
```

Comment:
setup does this step itself

Reply:
Noted.  I'll also see if I can get the branches straightened out.  I wasn't intending these commits to be part of this pull request.